### ISSUE TITLE
Normalise regions

### ISSUE THREAD

**Issue by dougaitken** [remoteintech/remote-jobs#485]
To allow for better matching and locations, we need to have predefined options for "region".

For example. For global reach, the default should be `Worldwide`. I made the first major change of this in #175 

As mentioned in #480, we don't have a defined option for United States / US / USA. 

I mainly use `US` but I'm open to discussing from Americans about what they'd use most often. 

If a company is limited to one country, then that should be listed, otherwise, we could use options like Worldwide, APAC, Europe, US (USA), Africa. 

I don't know what would be best / easiest. Either splitting via continent or region so I'd love to open this up for discussion

**Comment by Nirvikalpa108** [comment]
Hi @dougaitken and @nylen, thanks for starting this conversation which is really worthwhile. 

1) I agree that USA sounds good.

2) Do you think we could use the following convention for the region field? 
1 country = country
2+ countries in the same region = region
2+ regions = worldwide

So, for example, if the company has employees in 1 country, eg the UK, then we put UK in the region field. If they have employees in 2 countries, eg the UK and Germany, then we put Europe. If they have employees in the UK and US, then we put Worldwide. Does that make any sense?? Lol I don't know if I've just made it more complex! Let me know!

3) With regard to the naming of regions, whilst I have an intuitive feel as to which regions I would use, I decided to take a look at the [UN Statistical Devision's method](https://unstats.un.org/unsd/methodology/m49/), which is as follows:

Africa
Northern America
Latin America & the Caribbean
Asia
Europe
Oceania (this includes Australia)

I quite like this. Let me know what you think?

4) Also, @nylen I like your suggestion that we could put the region in the company profile and then repeat that in the Readme file. Have I understood you correctly? Please correct me if I'm wrong! Thank you guys

**Comment by nylen** [comment]
I quite like this. Let me know what you think?

**Comment by nylen** [comment]
Wow, sorry about the mis-clicks there.

> I quite like this. Let me know what you think?

I like it, except I think we should split out "Latin America" and "Caribbean" because together that is very long.

Let's simplify a bit further.

Region can be 1 country (USA), 1 region, or a list of multiple regions, or Worldwide.

If there are, let's say, 3 or more regions, then we'll keep the original list in the data here, but collapse this down to `Multiple` on the main page. The site can show the full list of regions on the company profile page.

And yes, we'll remove the Region column from the readme.

I think this is workable. In a little while I'll post a list of what I think needs to be done to make this happen.

**Comment by Nirvikalpa108** [comment]
Sounds good @nylen. Sorry if this is a really stupid question, but what's the difference between the main page and the readme that you've noted?
So, we're removing the Region column from the readme, but it's still on the main page and company profile?
Thanks 👍

**Comment by nylen** [comment]
The difference is that we can make any page of the site say whatever we want, as long as the data comes from _somewhere_. For now it mostly matches the format of the input files (main readme and company profiles), but it doesn't have to.

The site is automatically generated by code that parses the data out of the readme and the company profiles, and then reconstructs it in the format we want for the website. Again right now this format is basically the same as the input data, but it doesn't have to be.

In this case, we'll leave the region info in each company profile, clean it up for all profiles, and remove it from the readme, so that it only has to be written and updated once by us. Then we'll use that code to copy the region from each company profile to the main page of the site.

In short: it takes some preparatory work, but we can let the robots do the boring, repetitive tasks like copying data, as much as possible.

**Comment by Nirvikalpa108** [comment]
Nice nice, I understand now thank you! :smiley:

**Comment by nylen** [comment]
> I'll post a list of what I think needs to be done to make this happen

- Add some validation code (optional for now) to show a list of the companies that have a Region field that doesn't match the format above, or that have a different region between the readme and the company profile.  It's going to be a lot of errors and we'll need to fix them a few at a time.
- Fix the region field in all the company profiles.  For now we'll probably want to leave the region field in the readme, and make it the same as what's in the company profile.  This will also mean changing things like (for example) "UTC-8 to UTC-4" to "USA" or "North America".  Here it would be best to research a bit about each company, or ping the person who originally added the profile, but it doesn't have to be perfect.
- Decide what to do about the "extra" text in the Region field of the company profiles (I pulled all the values for Region in [my comment above](https://github.com/remoteintech/remote-jobs/issues/485#issuecomment-430484636), and there is a lot of extra stuff there).  We could allow this text to appear below the list of predefined regions, or we could move it to a different section of each profile.
- Once everything is ready, enable the validation code from the first step for all new PRs - this will make sure the data stays in good shape.
- Build a filter for region on the main page of the site.

**Comment by Nirvikalpa108** [comment]
Hi @nylen 
This looks good to me!

- Regarding the "extra" text in the Region field, do you think it's worth adding this to the "Remote Status" section of the company profile? I'm inclined to keep the text and just put it somewhere else in the profile rather than deleting, as some of it is quite useful.
- Also, do you think one of the items to add to your list is to change the Contributing Template and Company Profile template so that contributors know about the new rules for regions?

Thank you!:smiley:

**Comment by nylen** [comment]
Yes to both of those questions (move the extra text to the "Remote Status" field, and indicate the new rules for regions in our template files).

**Comment by nylen** [comment]
Sounds good. I'm thinking having the validation code in place, but leaving it as optional, would be useful in order to be able to generate a list of remaining issues whenever we want it. Are you able to run the existing validation code on your computer?

Definitely don't let that keep you from getting started though!

**Comment by dougaitken** [comment]
For context @Nirvikalpa108 - the readme list was the main list previously, James did the work to make it fancy and live on the website so we can move things away from that main list.

Ideally, everything lives in the company profile, we can then use that data if needed.

Thanks for your thoughts on this Amina!

**Comment by Nirvikalpa108** [comment]
Sounds great @nylen. I should be able to run the validation code. If I have any problems, I'll let you know, thank you 😃 
Thank you so much for giving me this context @dougaitken, you're a star! I really appreciate it! 😊

**Comment by Nirvikalpa108** [comment]
Hey guys, so here is a draft of the language I propose to put in the [Contributing Guidelines](remote-jobs/CONTRIBUTING.md) and the [Company Profiles - example.md](remote-jobs/company-profiles/example.md) pages so that contributors know what to put in the region field. If it's rubbish, no worries at all, it's just a first draft! I appreciate all your feedback! :smiley:

**[Contributing Guidelines](remote-jobs/CONTRIBUTING.md)** 
Number 5 - If you adhere to the protocol for listing the countries or regions for remote positions (as explained in the company profile - insert link)

_Do we want to remove the asterix saying that this is optional and make it a requirement? I think it should be a requirement given it would improve the site's functionality (eg the search function) and I think that's why this issue #485 exists, but let me know if I've misunderstood._

_Do you want to list the rule itself in the Contributing Guidelines or just link to it in the company profile page? Do you think contributors will follow the link and read it?_

**[Company Profiles - example.md](remote-jobs/company-profiles/example.md)**
Region
Where are remote employees accepted? List either 1 country (eg USA), 1 region (eg North America), a list of multiple regions, or Worldwide.

Here is the list of regions:
Africa
North America
Latin America
Caribbean
Asia
Europe
Oceania (this includes Australia)

If there are 3 or more regions they can be listed on the company profile page, but noted as Multiple on the main page. 

_Guys, this advice for the company profile page seems really long compared to the guidance for the other sections. This is just a first shot. Keen to get your thoughts. Once we're happy with wording I'll do a PR to submit changes. Thank you!:sunglasses:_

**Comment by Nirvikalpa108** [comment]
Hey guys, so here is a draft of the language I propose to put in the [Contributing Guidelines](remote-jobs/CONTRIBUTING.md) and the [Company Profiles - example.md](remote-jobs/company-profiles/example.md) pages so that contributors know what to put in the region field. If it's rubbish, no worries at all, it's just a first draft! I appreciate all your feedback! :smiley:

**[Contributing Guidelines](remote-jobs/CONTRIBUTING.md)** 
Number 5 - If you adhere to the protocol for listing the countries or regions for remote positions (as explained in the company profile - insert link)

_Do we want to remove the asterix saying that this is optional and make it a requirement? I think it should be a requirement given it would improve the site's functionality (eg the search function) and I think that's why this issue #485 exists, but let me know if I've misunderstood._

_Do you want to list the rule itself in the Contributing Guidelines or just link to it in the company profile page? Do you think contributors will follow the link and read it?_

**[Company Profiles - example.md](remote-jobs/company-profiles/example.md)**
Region
Where are remote employees accepted? List either 1 country (eg USA), 1 region (eg North America), a list of multiple regions, or Worldwide.

Here is the list of regions:
Africa
North America
Latin America
Caribbean
Asia
Europe
Oceania (this includes Australia)

If there are 3 or more regions they can be listed on the company profile page, but noted as Multiple on the main page. 

_Guys, this advice for the company profile page seems really long compared to the guidance for the other sections. This is just a first shot. Keen to get your thoughts. Once we're happy with wording I'll do a PR to submit changes. Thank you!:sunglasses:_

**Comment by nylen** [comment]
I don't think we should include the full details about this in the contributing guidelines.  We can link to them from there instead.  Also, our end goal here is that the automated check for the Region field is enabled and required, so let's make it required.

So, we could change the contributing guidelines from this:

> 5. If you submit the company with the regions that are accepted for remote positions (US only? Worldwide? Specific timezones?) *

to this:

> 5. If you submit the company with the regions that are accepted for remote positions (for more details [see the instructions in the example company profile](https://github.com/remoteintech/remote-jobs/blob/master/company-profiles/example.md#region)).

---

Don't forget the [pull request template](https://github.com/remoteintech/remote-jobs/blob/master/PULL_REQUEST_TEMPLATE.MD) text too.

There, we could just change this line in the pull request template:

> **Region** details any restrictions to applicants based on geography

to something like this:

> **Region** details the geographic regions in which this company's employees can reside.  For more details [see the instructions in the example company profile](https://github.com/remoteintech/remote-jobs/blob/master/company-profiles/example.md#region).

---

Now we need to fully document this new feature, and I think the example company profile is a good place to do that.  It's ok with me if this section is longer than the rest.  Here's a shot at it, continuing on from what you posted in your previous comment:

---

## Region

Where are remote employees accepted? List either 1 country (eg **USA**), 1 region (eg **North America**), a list of multiple regions, or **Worldwide**.

See the [UN Statistical Division's country codes](https://unstats.un.org/unsd/methodology/m49/) for a list of accepted countries.

The list of accepted regions is based on the UN Statistical Division's region codes, on the same page:

- Africa
- North America
- Latin America
- Caribbean
- Asia
- Europe
- Oceania _(this includes Australia)_

### Example region 1

USA

### Example region 2

North America, Latin America, Caribbean

### Example region 3

Worldwide

**Comment by Nirvikalpa108** [comment]
Thanks @nylen, this looks great to me! Happy to go ahead and make the PRs for this. Just want to check if @dougaitken has any changes? Thanks! 😃

**Comment by dougaitken** [comment]
Let's do this @Nirvikalpa108! We can tweak on the PR if needed

**Comment by Nirvikalpa108** [comment]
Great, thanks @dougaitken. PR submitted 😃 let me know if I've done anything wrong! Thanks!

**Comment by nylen** [comment]
#512 is an example of a PR where the region, as defined in our list above, is "Worldwide", but there is a bit of extra information:

> We prefer to work with remote team members in these timezones: UTC-8 to UTC-2.

In this case the extra text is in the "Remote status" section of the company profile.

I'm still happy with this approach, just pointing to a concrete example of what we discussed [above](https://github.com/remoteintech/remote-jobs/issues/485#issuecomment-431978970) and confirming that it will work well.

**Comment by Nirvikalpa108** [comment]
Great 👍

**Comment by dougaitken** [comment]
Hmm, that is a good point @nylen. I wonder if we could look at linking to the regions section of the profile and either using a plain link text or pulling in the data for the company.

Or maybe "specific countries"? if it isn't an obvious region match? 🤔

**Comment by nylen** [comment]
For the Hashicorp example above, and for others, it could work reasonably well to do a hybrid approach:

- Encourage specifying a well-structured list of countries as we have discussed here.  This can be a standard Markdown list.
- **Also allow free-form text below this list**.  This is the part that's new, and it would allow explaining any special cases, timezone requirements, etc.

This way we can let companies describe their region requirements accurately, but still have some pieces of structured data to enable a filter.

Here is another example where this approach could work well, but just listing "USA and India" wouldn't really be good enough (from #550):

![2019-04-10T21 19 00-05 00](https://user-images.githubusercontent.com/227022/55926344-93a98480-5bd6-11e9-8c8e-87791710eedd.png)

**Comment by dougaitken** [comment]
I thought the "example regions" which were in the Over submission might work well -https://github.com/remoteintech/remote-jobs/pull/549#pullrequestreview-225173210

Then there could be one shown, but for search, it could search the other regions?

**Comment by nylen** [comment]
Yeah, that works.  Those regions came from our [example profile](https://github.com/remoteintech/remote-jobs/blob/master/company-profiles/example.md) which is based on the previous discussion on this issue.

Let's get the data in place first, then we can figure out the details of how the search works.

**Comment by jainendra001** [comment]
Hi @dougaitken can I work on this issue, I am a full stack developer from India and currently an undergrad at IIT . Could you please assign it to me?

**Comment by dougaitken** [comment]
@jainendra001 As I mentioned in another Issue - before assigning any Issues, I would like possible contributors to discuss their solution in the Idea for enhancements. 

As you can see, this Issue has a long and complex history to try and get the "regions" to work well in the spirit of the list. Any solution will be fully discussed here before any PR should be opened.

**Comment by Dxuian** [comment]
is this still open ?

**Comment by dougaitken** [comment]
This topic is still in discussions. There have been improvements over the years but nothing solid to account for the varied ways companies hire in a distributed fashion.

**Comment by dougaitken** [comment]
Closing as completed - regions are now normalized in the Eleventy v3 site rebuild.

The site uses standardized region values defined in `src/_data/companyHelpers.js`:
- `worldwide`
- `americas`
- `europe`
- `americas-europe`
- `asia-pacific`
- `other`

Company profiles use these predefined values in their frontmatter, enabling consistent filtering and display.
